### DISCUSSION TITLE
Status Update: 2023-01-18

### DISCUSSION THREAD

**Discussion by haixuanTao** [dora-rs/dora#175] [Status Update]
:moon_cake: :partying_face:

**Comment by haixuanTao** [comment]
## dora

Push a new version `0.1.3` of dora with latest memory fixes for the `dora-drives`/`oasis` team.
Minor fixes such as:
- Better Version tagging management,
- Adding Changelog.md
- Adding python file into our dora python extension, to include python code.
- Fixed memory management issue after reviewing with the pyo3 maintainer. See: https://github.com/PyO3/pyo3/pull/2864 and https://github.com/dora-rs/dora/pulls/168
- Bump tokio version.

**Comment by haixuanTao** [comment]
## dora-drives

- Assisting `oasis` team in ramping up `dora-drives`. They are finally not using the `leaderboard 2.0` and reverting to `Carla 9.13` for the competition. They will replace the `leaderboard 2.0` with `oasis`. Reason being that they fear `leaderboard 2.0` and its Carla Version is not stable enough.

**Comment by haixuanTao** [comment]
## Partnership

- Reached back CentraleSupelec. Got in touch with the Head of Department of Computer Science at CentraleSupelec, and he supports our sponsoring of Master projects. We should normally have 1-2 Master projects starting Sept. 2023. He is happy to try both Hardware and Software projects.
- Still working on 1st-2nd Year academic program partnership.

**Comment by phil-opp** [comment]
## dora-daemon

- Create rudimentary shared memory server/client
  - Shared memory region + OS-provided condition variables for synchronization
  - Start creating a safe abstraction to make the code usable as separate crate
- Migrate the control and event channels used in `dora-daemon` from TCP sockets to the new shared memory server/client
  - Goal: Improve latency and throughput
- Refactor: Move shared memory handling to separate task
  - To reduce load on the main task

See the [`shared-mem-channel`](https://github.com/dora-rs/dora/commits/shared-mem-channel) branch for details.
